app/controllers/products_controller.rb

```diff
@@ -0,0 +1,16 @@
+class ProductsController < ApplicationController
+  before_action :set_product, only: [:show]
+  before_action :authenticate_user!
+
+  def index
+    @pagys, @products = pagy(Product.where(status: 'publish'), items: Product::PER_PAGE)
```

Comment:
If `status`. is an enum used `Product::STATUSES[:publish]`, otherwise create a constant and use over here. Do not use strings as standalone. Always use constants.